lib/matplotlib/tests/test_text.py

```diff
@@ -339,6 +339,28 @@ def test_set_position():
         assert a + shift_val == b
 
 
+def test_char_index_at():
+    fig = plt.figure()
+    text = fig.text(0.1, 0.9, "")
+
+    text.set_text("i")
+    bbox = text.get_window_extent()
+    size_i = bbox.x1 - bbox.x0
+
+    text.set_text("m")
+    bbox = text.get_window_extent()
+    size_m = bbox.x1 - bbox.x0
+
+    text.set_text("iiiimmmm")
+    bbox = text.get_window_extent()
+    origin = bbox.x0
+
+    assert text._char_index_at(origin) == 0
```

Comment:
Additional test for x left of first char
```suggestion
    assert text._char_index_at(origin - size_i) == 0  # left of first char
    assert text._char_index_at(origin) == 0
```